### ISSUE TITLE
Disable highlight-indet-guides-mode in swiper

### ISSUE THREAD

**Issue by CSRaghunandan** [DarthFennec/highlight-indent-guides#40]
Hello,

I wanted to know if it is possible to disable `highlight-indent-guides-mode` in `swiper`. swiper minibuffer would look a bit cleaner if it didn't include the indent guides :)

I'm not sure this can be done though. I tried adding `(highlight-indent-guides-mode -1)` to `minibuffer-setup-hook` but it did not work. There is no `swiper-mode-hook` available.

**Comment by jgkamat** [comment]
@CSRaghunandan See #38.

this mode is probably not enabled in swiper/ivy, it's probably the properties displayed from the copy instead. You could try stripping properties that are viewed in ivy (or on yank) if you don't want to see the guides there (`highlight-indent-guides-prop`)

**Comment by CSRaghunandan** [comment]
> this  mode is probably not enabled in swiper/ivy, it's probably the  properties displayed from the copy instead. You could try stripping  properties that are viewed in ivy (or on yank) if you don't want to see  the guides there (highlight-indent-guides-prop)

@jgkamat I'm an elisp noob. I have no idea how I can modify ivy for this. Can you pls write a small code snippet for this?

**Comment by jgkamat** [comment]
This is really hacky, but this works for me in `character` mode:

```
(defun jay/cleanup-hig-strings (x)
  (remove-text-properties 0 (length x) '(highlight-indent-guides-prop nil display) x))
(advice-add 'ivy-cleanup-string :after #'jay/cleanup-hig-strings)
```

It probably won't work in the others, you'd need to strip the faces as wel.

You could just also strip all faces.

**Comment by DarthFennec** [comment]
Yeah, swiper works by running font lock on the entire buffer in one pass, and then copying each line of the buffer into ivy, guides and all. @jgkamat's hack works because it removes the `display` properties in the copied strings, so character guides and partial tab guides disappear (just want to clarify that removing `highlight-indent-guides-prop` at this stage has no effect).

Fill and column guides exist as simple `face` properties, but just stripping all `face`s will remove all fontification in the swiper buffer, which is probably not what you want. It would be possible to write a cleanup function that only deletes specific faces, or only deletes faces on characters with a `highlight-indent-guides-prop` property, and leaves the rest untouched.

**Comment by seagle0128** [comment]
@DarthFennec Could you please give an example? Or add the snippet into FAQ? Thanks!

**Comment by DarthFennec** [comment]
@seagle0128 Sure thing, here's what I'm using:

``` emacs-lisp
(defadvice ivy-cleanup-string (after my-ivy-cleanup-hig activate)
  (let ((pos 0) (next 0) (limit (length str)) (prop 'highlight-indent-guides-prop))
    (while (and pos next)
      (setq next (text-property-not-all pos limit prop nil str))
      (when next
        (setq pos (text-property-any next limit prop nil str))
        (remove-text-properties next pos '(display nil face nil) str)))))
```

This will delete all `display` and `face` properties from any characters with a `highlight-indent-guides-prop` property, and it will leave the rest alone.

It could selectively only delete `h-i-g`-related faces, but doing that is probably slower, and it's unlikely that indentation is going to have other faces that you want to see in `swiper`. So I expect this is good enough.

This also doesn't check whether it's actually running in `swiper` or in another `ivy` client, so it can potentially cause slowness during other completion sessions. I doubt it would be in any noticeable amount, though, so I didn't bother.

**Comment by seagle0128** [comment]
@DarthFennec  Very nice! I am trying it and works well for now.
